### PR TITLE
SUBSYSTEMS.md: migrate subsystem list from wiki

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR migrate the [existing list of subsystems](https://github.com/RIOT-OS/RIOT/wiki/Maintainers) to the RIOT repo.
I also updated the list to represent the actual state of the system.

The list is definitely not perfect and may need several iterations. Feel free to add commits on demand.
We should definitely add more descriptions though, at least for the top headings (CPU, Core, etc).

The original entry included boards as well, but IMO it doesn't really make sense, as they are not a subsystem.

### Testing procedure
Read. Make sure that the list makes sense.

### Issues/PRs references
Discussed during the Virtual Maintainer Assembly

### CODE REVIEW

File: SUBSYSTEMS.md
Comment (benpicco) [inline SUBSYSTEMS.md:25]: Please don't just 'migrate' over inactive maintainers, start from scratch and let active maintainers fill in what they feel responsible for.

If we 'assign' people this will have to value, they have to actively feel responsible for their subsystem.

Comment (jia200x) [reply SUBSYSTEMS.md:25]: I kind of created the subsystems from scratch but migrated some of the maintainers. Should I just drop them all and let them add themselves?

Comment (maribu) [reply SUBSYSTEMS.md:25]: That sounds like a good idea. Let's also merge this without maintainers in it. We can distribute and parallize the work the better once the file is merged. (I fear otherwise there will be an endless discussion on nitpicking who to add and who not.)

Comment (benpicco) [reply SUBSYSTEMS.md:25]: Yes, we don't want the situation where people are like "Oh, I don't know why I'm on that list, I don't feel responsible for that subsystem".

There should be *some* commitment that can be expressed by actively editing the list.

I think we can also push to your branch directly.

Comment (jia200x) [reply SUBSYSTEMS.md]: done! I just removed all maintainers from the list. We can merge this one and then ask maintainers to add themselves to the list, as proposed above.

**Review by maribu — APPROVED** [review]
bors merge

**Pull request merged**
